docs/okeps/okep-5085-localnet-api.md

```diff
@@ -0,0 +1,825 @@
+
+The event should indicate pod network configuration is outdated, and suggest restarting to consume the latest network config.
+
+Comparing to annotation, events are harder to manage and get garbage collected.
+In addition, filtering events to realize which pod run outdated network-config is not intuitive 
+comparing to filtering by an annotation.
+
+Emitting events is a complementary matter for indicating the subject state, alongside the proposed solution.
+It can be done as a follow-up upon necessity and feedback from users.
+
+#### 2. Add condition to pod
+Similar to the idea of the proposed solution and option (1), but with adding status condition to the connected pods.
+
+Following CUDN CR localnet spec change, the CUDN controller should add a pod condition on the connected pods,
+providing indication about OVN-K network configuration state.
+The pod condition should be `true` when the pod network-config is up-to-date.
+The pod condition should be `false` when the pod network-config is outdated.
+
+Adding condition to pods is a complementary matter for indicating the subject state, alongside the proposed solution.
+It can be done as a follow-up upon necessity and feedback from users.
+
+#### 3. CUDN CRD status condition
+The CUDN CRD should have status condition indicating workloads configured with outdated network-configuration
+following localnet spec change.
+The condition should be `true` when there is at least one connected pod configured with the old spec.
+The condition should be `false` when all connected pods configured with the latest spec.
+
+The proposed solution serve as the baseline for providing the status condition, as the indication for when the 
+status condition should be true or false.
+
+Adding status condition is a complementary matter for indicating the subject state, alongside the proposed solution.
+It can be done as a follow-up upon necessity and feedback from users.
+
+#### 4. Provide collective data for alerting framework
+Alert is a notification that is raised based on various data sources in the cluster 
+such as object status, event, logs, metrics, etc..
+
+Usually alerting framework enable setting a notification on the platform GUI with a nice message referencing 
+to instructions on how to address it.
+
+The proposed pod annotation can serve as the baseline for providing collective data for alerting frameworks,
+as the indication for raising an alert, e.g.: when there is at least on pod with outdated network-config.
+
+Alerts are a complementary matter for indicating the subject state, alongside the proposed solution.
+It can be done as a follow-up upon necessity and feedback from users.
+
+#### 5. Add support for OVN-K CNI plugin to configure running pods  
+As of today OVN-K configure the pod network-stack when a pod is created or deleted.
+
+Introduce OVN-K CNI plugin using a think plugin architecture, similar to Multus CNI think-plugin
```

Comment:
you mean "thick" plugin?

we already have something similar with our CNI server. We can detect changes to the NAD there. Alternatively we could implement cni cmd CHECK, that would allow us to reconfigure something in the pod. Anyway, this is something that we can tackle in the future.

Reply:
Yeah, sorry for this occurred typo.. it was late :see_no_evil: 

> we already have something similar with our CNI server. We can detect changes to the NAD there. Alternatively we could implement cni cmd CHECK, that would allow us to reconfigure something in the pod. Anyway, this is something that we can tackle in the future.

That sound really useful, in particular for MTU changes, which have even greater impact on VM (as we strive for minimal downtime).